controller/instance_handler.go

```diff
@@ -226,9 +226,10 @@ func (h *InstanceHandler) ReconcileInstanceState(obj interface{}, spec *types.In
 				logrus.Warnf("Try to get requested log for %v on node %v", instanceName, im.Spec.NodeID)
 				if err := h.printInstanceLogs(instanceName, runtimeObj); err != nil {
 					logrus.Warnf("cannot get requested log for instance %v on node %v, error %v", instanceName, im.Spec.NodeID, err)
+				} else {
+					status.LogFetched = true
 				}
 			}
-			status.LogFetched = true
```

Comment:
We need to consider the case described in the comment https://github.com/longhorn/longhorn-manager/pull/959#issuecomment-876978286

Reply:
See PR comment, we can move the setting of r.spec.LogRequested into the same guard then when the replica fails.
SInce there is no need to continously request the logs in this case once the replica has failed @shuo-wu correct?

Reply:
@PhanLe1010 NIT: can you see some ways to simplify the above if/else structure?

Reply:
Done. Thank you